### PR TITLE
#834 : support for --help

### LINKED ISSUE #834: support for --help
Common CLIs should support `--help` option according to Posix-Standard.
Currently to get the help of IDEasy, you need to call `ide help` or `ideasy help` but `ide --help` is not working.
This should be changed.
Ideally also `-h` should be supported (however, currently we only support a single alias).

### PR DESCRIPTION
**Pull request by hohwille**
fixes #834

Allow commandlet/keyword to use both `«name»` as well as `--«name»` and an additional short options als alias `-«a»`.
Apply the new feature to both `VersionCommandlet` and `HelpCommandlet`.

### CODE REVIEW

File: cli/src/main/java/com/devonfw/tools/ide/property/Property.java [hunk 1]
```diff
@@ -331,13 +331,25 @@ protected V getNullValue() {
...
+  }
+
+  /**
+   * @param args the {@link CliArguments} already {@link CliArguments#current() pointing} the {@link CliArgument} to apply.
+   * @param context the {@link IdeContext}.
+   * @param commandlet the {@link Commandlet} owning this property.
+   * @param collector the {@link CompletionCandidateCollector}.
+   * @return {@code true} if it matches, {@code false} otherwise.
+   */
+  protected boolean apply(String normalizedName, CliArguments args, IdeContext context, Commandlet commandlet, CompletionCandidateCollector collector) {
```
Comment (jan-vcapgemini) [inline cli/src/main/java/com/devonfw/tools/ide/property/Property.java:344]: Please adjust Javadoc. normalizedName is missing.

File: cli/src/main/java/com/devonfw/tools/ide/property/Property.java [hunk 2]
```diff
@@ -404,19 +416,20 @@ protected boolean applyValue(String argValue, boolean lookahead, CliArguments ar
    * @param commandlet the {@link Commandlet} owning this {@link Property}.
    * @param collector the {@link CompletionCandidateCollector}.
    */
-  protected void complete(CliArgument argument, CliArguments args, IdeContext context, Commandlet commandlet, CompletionCandidateCollector collector) {
+  protected void complete(String nameNormalized, CliArgument argument, CliArguments args, IdeContext context, Commandlet commandlet,
+      CompletionCandidateCollector collector) {
```
Comment (jan-vcapgemini) [inline cli/src/main/java/com/devonfw/tools/ide/property/Property.java:420]: Please adjust Javadoc, normalizedName is missing.

**Review by jan-vcapgemini — APPROVED** [review]
LGTM. I've added some small CRs please resolve.

File: cli/src/main/java/com/devonfw/tools/ide/property/Property.java (hunk 1 above)
Comment (hohwille) [reply cli/src/main/java/com/devonfw/tools/ide/property/Property.java]: Done.

File: cli/src/main/java/com/devonfw/tools/ide/property/Property.java (hunk 2 above)
Comment (hohwille) [reply cli/src/main/java/com/devonfw/tools/ide/property/Property.java]: Done. Also aligned the parameter name (`nameNormalized` vs. `normalizedName`). Also found another undocumented method from someone else and added JavaDoc.

**Pull request merged**